src/api/nodeRpcCall.js

```diff
@@ -774,37 +766,24 @@ const submitProposal = async (walletAddress, values, callback) => {
     additionalMetadata: {},
     proposerAddress: walletAddress,
   });
-  const legislationIndex = centralizedMetadata.data.id;
-  const hash = await getProposalHash(values, legislationIndex);
-  const notePreimageTx = api.tx.preimage.notePreimage(hash.extrinsicEncoded);
+  const extrinsicMethod = api.tx.liberlandLegislation.addLaw(
```

Comment:
```suggestion
  const addLaw = api.tx.liberlandLegislation.addLaw(
```
`extrinsic` is the whole transaction. And is a bit too generic, given that we also operate on `democracy.propose*`, `preimage` and `utility.batchAll` calls